generation_scripts/abiotic/climate_data_processing_script.py

```diff
@@ -0,0 +1,227 @@
+#       Ready for ingestion by the VE data loader.
+#
+# package_dependencies:
+#   - numpy
+#   - xarray
+#   - tomllib (Python 3.11+)
+#   - pyproj
+#
+# usage_notes: |
+#   - Uses nearest-neighbour interpolation (xarray.interp with method='nearest').
+#   - EPSG code in the TOML must be valid for pyproj (e.g. 32650).
+#   - Precipitation conversion assumes 30-day months (tp * 1000 * 30).
+#   - Spatial downscaling/topographic correction is not yet implemented.
+# ---
+
+
+import numpy as np
+import tomllib
+import xarray as xr
+from pyproj import Transformer
+
+# ---------------------------
+# File paths (edit here)
+# ---------------------------
+toml_file = (
+    r"C:....ve_data_analysis\abiotic\data_download\site\maliau_grid_definition.toml"
+)
+era5_file = (
+    r"C:.....ve_data_science\data\primary\abiotic\era5_monthly_2010_2020_maliau.nc"
+)
+output_file = (
+    r"C:....ve_data_science\data\derived\abiotic\processed_climate_data_maliau.nc"
+)
+
+# ---------------------------
+# 1. Load TOML site definition
+# ---------------------------
+# The TOML file is **site-specific**. It provides:
+#   - Projected grid cell centres (cell_x, cell_y)
+#   - EPSG code for the coordinate reference system (e.g., UTM zone)
+#   - WGS84 bounds (lat/lon extent of the study area)
+#   - Target resolution for VE grid cells
+# Each site should have its own TOML definition for correct alignment.
+
+with open(toml_file, "rb") as f:
+    site_config = tomllib.load(f)
+
+bounds_wgs84 = site_config["wgs84_bounds"]  # [lat_min, lon_min, lat_max, lon_max]
+epsg_code = site_config["epsg_code"]
+res = site_config["res"]
+
+cell_x = np.array(site_config["cell_x_centres"])
+cell_y = np.array(site_config["cell_y_centres"])
+
+# Build 2D meshgrid in projected CRS
+x_grid, y_grid = np.meshgrid(cell_x, cell_y)
+
+# Convert projected UTM grid → WGS84 lat/lon
+transformer = Transformer.from_crs(epsg_code, 4326, always_xy=True)
+lon_grid, lat_grid = transformer.transform(x_grid, y_grid)
+
+# ---------------------------
+# 2. Load ERA5-Land dataset
+# ---------------------------
+# This ERA5-Land dataset covers a large area of grid cells at a coarser resolution (~9 km).
+
+dataset = xr.open_dataset(era5_file)
+
+if "valid_time" in dataset.dims or "valid_time" in dataset.coords:
+    dataset = dataset.rename({"valid_time": "time"})
+
+# ---------------------------
+# 3. Unit conversions
+# ---------------------------
+# The standard output unit of ERA5-Land temperatures is Kelvin which we need to convert
+# to degree Celsius for the Virtual Ecosystem. This includes 2m air temperature and
+# 2m dewpoint temperature which are used to calculate relative humidity in next step.
+dataset["t2m_C"] = dataset["t2m"] - 273.15
+dataset["d2m_C"] = dataset["d2m"] - 273.15
+
+# Relative humidity (RH) is not a standard output from ERA5-Land but can be calculated
+# from 2m dewpoint temperature (DPT) and 2m air temperature (T)
```

Comment:
I think the abbreviations in brackets are relicts from the earlier script, I don't thin they are needed anymore. It would be useful to have the units instead